--- a/src/sound/ymfm/ymfm_opn.h
+++ b/src/sound/ymfm/ymfm_opn.h
@@ -763,7 +763,7 @@
 
 protected:
 	// simulate the DAC discontinuity
-	constexpr int32_t dac_discontinuity(int32_t value) const { return (value < 0) ? (value - 3) : (value + 4); }
+	static constexpr int32_t dac_discontinuity(int32_t value) { return (value < 0) ? (value - 3) : (value + 4); }
 
 	// internal state
 	uint16_t m_address;              // address register
