--- aircrack-ng-1.2-rc4.org/scripts/airmon-ng
+++ aircrack-ng-1.2-rc4/scripts/airmon-ng
@@ -490,9 +490,9 @@
 	setLink ${1} up
 	if [ -n "${CH}" ]; then
 		if [ ${CH} -lt 1000 ]; then
-			#mac80211 specific check to see if hardware supports the channel number being assigned (requires grep with pcre)
+			#mac80211 specific check to see if hardware supports the channel number being assigned
 			if [ -r "/sys/class/net/$1/phy80211/name" ]; then
-		                channel_list="$(iw phy $(cat /sys/class/net/$1/phy80211/name) info 2>&1 | grep -oP '\[\K[^\]]+')"
+		                channel_list="$(iw phy $(cat /sys/class/net/$1/phy80211/name) info 2>&1 | awk '/\[.*\]/{print substr($4,2,length($4)-2)}')"
 				local hardware_valid_channel=1
 	        	        for i in $channel_list; do
 	                	        if [ "${CH}" = "${i}" ]; then
@@ -507,9 +507,9 @@
 			fi
 			IW_ERROR="$(iw dev ${1} set channel ${CH} 2>&1)"
 		else
-			#mac80211 specific check to see if hardware supports the frequency number being assigned (requires grep with pcre)
+			#mac80211 specific check to see if hardware supports the frequency number being assigned
 			if [ -r "/sys/class/net/$1/phy80211/name" ]; then
-				frequency_list="$(iw phy $(cat /sys/class/net/$1/phy80211/name) info 2>&1 | grep -oP '\*\ \K[\d]+\ ')"
+				frequency_list="$(iw phy $(cat /sys/class/net/$1/phy80211/name) info 2>&1 | awk '/\[.*\]/{print $2}')"
 				local hardware_valid_freq=1
 				for i in $frequency_list; do
 					if [ "${CH}" = "${i}" ]; then
