https://github.com/aircrack-ng/aircrack-ng/pull/126

--- aircrack-ng-1.2-rc1.org/src/airodump-ng.c
+++ aircrack-ng-1.2-rc1/src/airodump-ng.c
@@ -477,7 +477,7 @@
     if( time(NULL) - G.sk_start > 5)
     {
         /* timeout(5sec) - remove all packets, restart timer */
-        memset(G.sharedkey, '\x00', 4096*3);
+        memset(G.sharedkey, '\x00', sizeof(G.sharedkey));
         G.sk_start = time(NULL);
     }
 
@@ -537,7 +537,7 @@
     }
 
     textlen = G.sk_len;
-
+#if 0
     if(textlen+4 != G.sk_len2)
     {
         snprintf(G.message, sizeof(G.message), "][ Broken SKA: %02X:%02X:%02X:%02X:%02X:%02X ",
@@ -545,7 +545,7 @@
                 *(G.sharedkey[0]+m_bmac+3), *(G.sharedkey[0]+m_bmac+4), *(G.sharedkey[0]+m_bmac+5));
         return 1;
     }
-
+#endif
     if((unsigned)textlen > sizeof(text) - 4) return 1;
 
     memcpy(text, G.sharedkey[0]+24, textlen);
@@ -607,7 +607,7 @@
                 textlen+4, *(G.sharedkey[0]+m_bmac), *(G.sharedkey[0]+m_bmac+1), *(G.sharedkey[0]+m_bmac+2),
               *(G.sharedkey[0]+m_bmac+3), *(G.sharedkey[0]+m_bmac+4), *(G.sharedkey[0]+m_bmac+5));
 
-    memset(G.sharedkey, '\x00', 512*3);
+    memset(G.sharedkey, '\x00', sizeof(G.sharedkey));
     /* ok, keystream saved */
     return 0;
 }
@@ -5652,7 +5652,7 @@
 	// Default selection.
     resetSelection();
 
-    memset(G.sharedkey, '\x00', 512*3);
+    memset(G.sharedkey, '\x00', sizeof(G.sharedkey));
     memset(G.message, '\x00', sizeof(G.message));
     memset(&G.pfh_in, '\x00', sizeof(struct pcap_file_header));
 
