--- aria2-1.19.0.org/src/SocketCore.cc
+++ aria2-1.19.0/src/SocketCore.cc
@@ -33,6 +33,7 @@
  */
 /* copyright --> */
 #include "SocketCore.h"
+#include "LogFactory.h"
 
 #ifdef HAVE_IPHLPAPI_H
 # include <iphlpapi.h>
@@ -630,7 +631,7 @@
 #ifndef __MINGW32__
 # define CHECK_FD(fd)                                                   \
   if(fd < 0 || FD_SETSIZE <= fd) {                                      \
-    logger_->warn("Detected file descriptor >= FD_SETSIZE or < 0. "     \
+    A2_LOG_WARN("Detected file descriptor >= FD_SETSIZE or < 0. "     \
                   "Download may slow down or fail.");                   \
     return false;                                                       \
   }
