--- busybox-1.33.0.org/util-linux/fdisk.c	2021-01-27 05:41:05.320303095 +0000
+++ busybox-1.33.0/util-linux/fdisk.c	2021-01-27 05:42:16.374299684 +0000
@@ -164,17 +164,8 @@
 
 
 typedef unsigned long long ullong;
-/* Used for sector numbers. Partition formats we know
- * do not support more than 2^32 sectors
- */
-typedef uint32_t sector_t;
-#if UINT_MAX == 0xffffffff
-# define SECT_FMT ""
-#elif ULONG_MAX == 0xffffffff
-# define SECT_FMT "l"
-#else
-# error Cant detect sizeof(uint32_t)
-#endif
+typedef uint64_t sector_t;
+#define SECT_FMT "ll"
 
 struct hd_geometry {
 	unsigned char heads;
