--- a/libbb/procps.c
+++ b/libbb/procps.c
@@ -473,11 +473,17 @@
 				sp->state[1] = 'W';
 			else
 				sp->state[1] = ' ';
-			if (tasknice < 0)
-				sp->state[2] = '<';
-			else if (tasknice) /* > 0 */
-				sp->state[2] = 'N';
-			else
+			if (tasknice < 0) {
+				if(sp->state[1] == ' ') {
+					sp->state[1] = '<';
+					sp->state[2] = ' ';
+				} else sp->state[2] = '<';
+			} else if (tasknice) /* > 0 */ {
+				if(sp->state[1] == ' ') {
+					sp->state[1] = 'N';
+					sp->state[2] = ' ';
+				} else 	sp->state[2] = 'N';
+			} else
 				sp->state[2] = ' ';
 		}
 
