diff --git a/netbsd_sys/endian.h b/netbsd_sys/endian.h
index a79193b..dd9d693 100644
--- a/netbsd_sys/endian.h
+++ b/netbsd_sys/endian.h
@@ -18,18 +18,16 @@
 #define le32toh(x) (uint32_t)(x)
 #define le16toh(x) (uint16_t)(x)
 #else
-#ifndef __bswap32
-static __inline uint32_t __bswap32(uint32_t __x)
+static __inline uint32_t nbs_bswap32(uint32_t __x)
 {
 	return __x>>24 | __x>>8&0xff00 | __x<<8&0xff0000 | __x<<24;
 }
-static __inline uint16_t __bswap16(uint16_t __x)
+static __inline uint16_t nbs_bswap16(uint16_t __x)
 {
 	return __x<<8 | __x>>8;
 }
-#endif
-#define le32toh(x) __bswap32(x)
-#define le16toh(x) __bswap16(x)
+#define le32toh(x) nbs_bswap32(x)
+#define le16toh(x) nbs_bswap16(x)
 #endif
 #endif
 
-- 
2.20.1

