--- dolphin-5.0.org/Source/Core/Common/Thread.cpp
+++ dolphin-5.0/Source/Core/Common/Thread.cpp
@@ -8,6 +8,7 @@
 
 #ifndef _WIN32
 #include <unistd.h>
+#include <string.h>
 #endif
 
 #ifdef __APPLE__
--- dolphin-5.0.org/Source/Core/Core/MemoryWatcher.cpp
+++ dolphin-5.0/Source/Core/Core/MemoryWatcher.cpp
@@ -7,6 +7,7 @@
 #include <memory>
 #include <sstream>
 #include <unistd.h>
+#include <string.h>
 
 #include "Common/FileUtil.h"
 #include "Core/CoreTiming.h"
