--- mozilla-esr38/gfx/skia/trunk/src/core/SkUtilsArm.cpp.org
+++ mozilla-esr38/gfx/skia/trunk/src/core/SkUtilsArm.cpp
@@ -47,6 +47,15 @@
 #  endif
 #endif
 
+#ifndef TEMP_FAILURE_RETRY
+#define TEMP_FAILURE_RETRY(expression) \
+  (__extension__                                  \
+    ({ long int __result;                             \
+       do __result = (long int) (expression);                     \
+       while (__result == -1L && errno == EINTR);                 \
+       __result; }))
+#endif
+
 // A function used to determine at runtime if the target CPU supports
 // the ARM NEON instruction set. This implementation is Linux-specific.
 static bool sk_cpu_arm_check_neon(void) {
