--- libressl-2.9.2/crypto/compat/getprogname_linux.c
+++ libressl-2.9.2.org/crypto/compat/getprogname_linux.c
@@ -23,7 +23,7 @@
 	 *
 	 * .. [3] https://android.googlesource.com/platform/system/core/+/2819c0/base/logging.cpp#65
 	 */
-#if defined(__ANDROID_API__) && __ANDROID_API__ < 21
+#if defined(HAVE___PROGNAME) || (defined(__ANDROID_API__) && __ANDROID_API__ < 21)
 	extern const char *__progname;
 	return __progname;
 #elif defined(__GLIBC__)
