--- nano-2.7.2.org/src/text.c
+++ nano-2.7.2/src/text.c
@@ -3829,11 +3829,13 @@
 	    do_output(&completion[shard_length],
 			strlen(completion) - shard_length, FALSE);
 
+#ifndef DISABLE_WRAPPING
 	    /* If needed, reenable wrapping and wrap the current line. */
 	    if (was_set_wrapping) {
 		UNSET(NO_WRAP);
 		do_wrap(openfile->current);
 	    }
+#endif
 
 	    /* Set the position for a possible next search attempt. */
 	    pletion_x = ++i;
