--- openssh-8.4p1.org/sshconnect.c
+++ openssh-8.4p1/sshconnect.c
@@ -755,10 +755,13 @@
 
 	/*
 	 * Turn off check_host_ip if the connection is to localhost, via proxy
-	 * command or if we don't have a hostname to compare with
+	 * command or if we don't have a hostname to compare with, or
+	 * if PROXYCHAINS-NG is active and the ip starts with "224."
+	 * (default remote-dns subnet).
 	 */
 	if (options.check_host_ip && (local ||
-	    strcmp(hostname, ip) == 0 || options.proxy_command != NULL))
+	    strcmp(hostname, ip) == 0 || options.proxy_command != NULL ||
+	    (strncmp(ip, "224.", 4) == 0 && getenv("PROXYCHAINS_CONF_FILE"))))
 		options.check_host_ip = 0;
 
 	host_hostkeys = init_hostkeys();
