--- pale-moon.org/platform/netwerk/protocol/websocket/WebSocketChannel.h
+++ pale-moon/platform/netwerk/protocol/websocket/WebSocketChannel.h
@@ -146,6 +146,7 @@
   void GeneratePong(uint8_t *payload, uint32_t len);
   void GeneratePing();
 
+  nsresult OnNetworkChangedTargetThread();
   nsresult OnNetworkChanged();
   nsresult StartPinging();
 
--- pale-moon.org/platform/netwerk/protocol/websocket/WebSocketChannel.cpp
+++ pale-moon/platform/netwerk/protocol/websocket/WebSocketChannel.cpp
@@ -1247,10 +1247,11 @@
         // Next we check mDataStarted, which we need to do on mTargetThread.
         if (!IsOnTargetThread()) {
           mTargetThread->Dispatch(
-            NewRunnableMethod(this, &WebSocketChannel::OnNetworkChanged),
+            NewRunnableMethod(this,
+            &WebSocketChannel::OnNetworkChangedTargetThread),
             NS_DISPATCH_NORMAL);
         } else {
-          OnNetworkChanged();
+          OnNetworkChangedTargetThread();
         }
       }
     }
@@ -1260,21 +1261,24 @@
 }
 
 nsresult
-WebSocketChannel::OnNetworkChanged()
+WebSocketChannel::OnNetworkChangedTargetThread()
 {
-  if (IsOnTargetThread()) {
-    LOG(("WebSocketChannel::OnNetworkChanged() - on target thread %p", this));
+  LOG(("WebSocketChannel::OnNetworkChangedTargetThread() - on target thread %p", this));
 
-    if (!mDataStarted) {
-      LOG(("WebSocket: data not started yet, no ping needed"));
-      return NS_OK;
-    }
-
-    return mSocketThread->Dispatch(
-      NewRunnableMethod(this, &WebSocketChannel::OnNetworkChanged),
-      NS_DISPATCH_NORMAL);
+  if (!mDataStarted) {
+    LOG(("WebSocket: data not started yet, no ping needed"));
+    return NS_OK;
   }
 
+  return mSocketThread->Dispatch(
+    NewRunnableMethod(this,
+                      &WebSocketChannel::OnNetworkChanged),
+    NS_DISPATCH_NORMAL);
+}
+
+nsresult
+WebSocketChannel::OnNetworkChanged()
+{
   MOZ_ASSERT(PR_GetCurrentThread() == gSocketThread, "not socket thread");
 
   LOG(("WebSocketChannel::OnNetworkChanged() - on socket thread %p", this));
