--- psmisc-23.2.orig/src/peekfd.c	2019-08-17 23:20:02.951223553 +0000
+++ psmisc-23.2/src/peekfd.c	2019-08-18 01:05:09.191249051 +0000
@@ -27,10 +27,14 @@
 #include <sys/types.h>
 #include <sys/wait.h>
 #include <sys/syscall.h>
+#include <sys/user.h>
+#ifdef __powerpc__
+#define pt_regs asdf
+#endif
 #include <asm/ptrace.h>
+#undef pt_regs
 #include <byteswap.h>
 #include <endian.h>
-#include <sys/user.h>
 #include <stdlib.h>
 #include <getopt.h>
 #include <ctype.h>
