--- pwntools-3.6.1.org/pwnlib/elf/elf.py
+++ pwntools-3.6.1/pwnlib/elf/elf.py
@@ -609,7 +609,7 @@
             return
 
         try:
-            cmd = 'ulimit -s unlimited; LD_TRACE_LOADED_OBJECTS=1 LD_WARN=1 LD_BIND_NOW=1 %s 2>/dev/null' % sh_string(self.path)
+            cmd = 'ulimit -s unlimited; ldd %s 2>/dev/null' % sh_string(self.path)
 
             data = subprocess.check_output(cmd, shell = True, stderr = subprocess.STDOUT)
             libs = misc.parse_ldd_output(data)
