--- radare2-0.10.0/binr/rabin2/Makefile
+++ radare2-0.10.0.patched/binr/rabin2/Makefile
@@ -3,6 +3,7 @@
 BINDEPS+=r_config r_io r_flags r_asm
 BINDEPS+=r_debug r_hash r_lang r_anal r_parse r_bp r_reg
 BINDEPS+=r_search r_syscall r_socket r_fs r_magic r_util
+LDFLAGS+=-lcapstone
 
 include ../rules.mk
 include ../../libr/socket/deps.mk
--- radare2-0.10.0/binr/radare2/Makefile
+++ radare2-0.10.0.patched/binr/radare2/Makefile
@@ -3,7 +3,7 @@
 BINDEPS+=r_bin r_debug r_anal r_reg r_bp r_io r_fs
 BINDEPS+=r_lang r_asm r_syscall r_db r_hash
 BINDEPS+=r_magic r_socket r_flags r_util r_egg r_crypto
-
+LDFLAGS+=-lcapstone
 # Apple Sign #
 CERTID=org.radare.radare2
 BUNDLE=org.radare.radare2
--- radare2-0.10.0/binr/radiff2/Makefile
+++ radare2-0.10.0.patched/binr/radiff2/Makefile
@@ -9,6 +9,7 @@
 ifeq ($(OSTYPE),android)
 LDFLAGS+=${DL_LIBS} -lm
 endif
+LDFLAGS+=-lcapstone
 
 include ../../libr/socket/deps.mk
 include ../../shlr/zip/deps.mk
--- radare2-0.10.0/binr/rafind2/Makefile
+++ radare2-0.10.0.patched/binr/rafind2/Makefile
@@ -1,6 +1,7 @@
 BIN=rafind2
 BINDEPS=r_search r_io r_asm r_anal r_reg r_cons
-BINDEPS+=r_socket r_db r_parse r_util r_syscall r_flags
+BINDEPS+=r_socket r_db r_parse r_util r_syscall r_flags r_lang
+LDFLAGS+=-lcapstone
 
 include ../rules.mk
 
--- radare2-0.10.0/binr/ragg2/Makefile
+++ radare2-0.10.0.patched/binr/ragg2/Makefile
@@ -1,6 +1,6 @@
 BIN=ragg2
 BINDEPS=r_egg r_bin r_syscall r_asm r_db r_parse r_flags r_magic
-BINDEPS+=r_anal r_reg r_cons r_io r_util r_socket
-
+BINDEPS+=r_anal r_reg r_cons r_io r_util r_socket r_lang
+LDFLAGS+=-lcapstone
 include ../rules.mk
 include ../../libr/socket/deps.mk
--- radare2-0.10.0/binr/rahash2/Makefile
+++ radare2-0.10.0.patched/binr/rahash2/Makefile
@@ -1,13 +1,14 @@
 BIN=rahash2
 BINDEPS=r_io r_hash r_socket r_asm r_cons r_util
 BINDEPS+=r_anal r_syscall r_reg r_db r_parse
-BINDEPS+=r_flags
+BINDEPS+=r_flags r_lang
 
 include ../rules.mk
 
 ifeq ($(OSTYPE),android)
 LDFLAGS+=-lm
 endif
+LDFLAGS+=-lcapstone
 include ../../libr/socket/deps.mk
 include ../../shlr/zip/deps.mk
 include ../../shlr/gdb/deps.mk
--- radare2-0.10.0/binr/rasm2/Makefile
+++ radare2-0.10.0.patched/binr/rasm2/Makefile
@@ -1,4 +1,5 @@
 BIN=rasm2
-BINDEPS=r_asm r_parse r_syscall r_anal r_reg r_db r_flags r_cons r_util
+BINDEPS=r_asm r_parse r_syscall r_anal r_reg r_db r_flags r_cons r_util r_lang
+LDFLAGS+=-lcapstone
 
 include ../rules.mk
