From 85034079acf8603d0d9af68ae15c4a799e489cfb Mon Sep 17 00:00:00 2001
From: rofl0r <rofl0r@users.noreply.github.com>
Date: Sat, 11 Sep 2021 11:47:46 +0000
Subject: [PATCH] revert 192606537759c3e

---
 src/asm/parser.y | 32 +++++++++++++++++++++-----------
 1 file changed, 21 insertions(+), 11 deletions(-)

diff --git a/src/asm/parser.y b/src/asm/parser.y
index fe69ab4..a3a9ce0 100644
--- a/src/asm/parser.y
+++ b/src/asm/parser.y
@@ -509,6 +509,7 @@ enum {
 %token	T_LBRACK "[" T_RBRACK "]"
 %token	T_LPAREN "(" T_RPAREN ")"
 %token	T_NEWLINE "newline"
+%left	T_COMMA T_COLON T_LBRACK T_RBRACK T_LPAREN T_RPAREN T_NEWLINE
 
 %token	T_OP_LOGICNOT "!"
 %token	T_OP_LOGICAND "&&" T_OP_LOGICOR "||"
@@ -520,6 +521,7 @@ enum {
 %token	T_OP_SHL "<<" T_OP_SHR ">>"
 %token	T_OP_MUL "*" T_OP_DIV "/" T_OP_MOD "%"
 %token	T_OP_NOT "~"
+%left	T_OP_LOGICNOT
 %left	T_OP_LOGICOR
 %left	T_OP_LOGICAND
 %left	T_OP_LOGICGT T_OP_LOGICLT T_OP_LOGICGE T_OP_LOGICLE T_OP_LOGICNE T_OP_LOGICEQU
@@ -527,8 +529,9 @@ enum {
 %left	T_OP_OR T_OP_XOR T_OP_AND
 %left	T_OP_SHL T_OP_SHR
 %left	T_OP_MUL T_OP_DIV T_OP_MOD
+%left	T_OP_NOT
 
-%precedence	NEG /* negation -- unary minus */
+%left	NEG /* negation -- unary minus */
 
 %token	T_OP_EXP "**"
 %left	T_OP_EXP
@@ -545,6 +548,10 @@ enum {
 %token	T_OP_LOG "LOG"
 %token	T_OP_ROUND "ROUND"
 %token	T_OP_CEIL "CEIL" T_OP_FLOOR "FLOOR"
+%left	T_OP_DEF T_OP_BANK T_OP_ALIGN
+%left	T_OP_SIN T_OP_COS T_OP_TAN T_OP_ASIN T_OP_ACOS T_OP_ATAN T_OP_ATAN2
+%left	T_OP_FDIV T_OP_FMUL T_OP_POW T_OP_LOG
+%left	T_OP_ROUND T_OP_CEIL T_OP_FLOOR
 
 %token	T_OP_HIGH "HIGH" T_OP_LOW "LOW"
 %token	T_OP_ISCONST "ISCONST"
@@ -557,6 +564,8 @@ enum {
 %token	T_OP_STRUPR "STRUPR" T_OP_STRLWR "STRLWR"
 %token	T_OP_STRRPL "STRRPL"
 %token	T_OP_STRFMT "STRFMT"
+%left	T_OP_STRCMP T_OP_STRIN T_OP_STRRIN T_OP_STRSUB T_OP_STRLEN T_OP_STRCAT
+%left	T_OP_STRUPR T_OP_STRLWR T_OP_STRRPL T_OP_STRFMT
 
 %token	T_OP_CHARLEN "CHARLEN"
 %token	T_OP_CHARSUB "CHARSUB"
@@ -668,7 +677,7 @@ enum {
 asmfile		: lines
 ;
 
-lines		: %empty
+lines		: /* empty */
 		| lines line
 ;
 
@@ -791,7 +800,7 @@ redef_id	: T_POP_REDEF {
 scoped_id	: T_ID | T_LOCAL_ID;
 scoped_anon_id	: scoped_id | T_ANON;
 
-label		: %empty
+label		: /* empty */
 		| T_COLON {
 			sym_AddAnonLabel();
 		}
@@ -822,7 +831,7 @@ macro		: T_ID {
 		}
 ;
 
-macroargs	: %empty {
+macroargs	: /* empty */ {
 			$$ = macro_NewArgs();
 		}
 		| macroargs T_STRING {
@@ -885,7 +894,8 @@ directive	: endc
 		| align
 ;
 
-trailing_comma	: %empty | T_COMMA
+trailing_comma	: /* empty */
+		| T_COMMA
 ;
 
 equ		: T_LABEL T_POP_EQU const { sym_AddEqu($1, $3); }
@@ -966,7 +976,7 @@ fail		: T_POP_FAIL string { fatalerror("%s\n", $2); }
 warn		: T_POP_WARN string { warning(WARNING_USER, "%s\n", $2); }
 ;
 
-assert_type	: %empty { $$ = ASSERT_ERROR; }
+assert_type	: /* empty */ { $$ = ASSERT_ERROR; }
 		| T_POP_WARN T_COMMA { $$ = ASSERT_WARN; }
 		| T_POP_FAIL T_COMMA { $$ = ASSERT_ERROR; }
 		| T_POP_FATAL T_COMMA { $$ = ASSERT_FATAL; }
@@ -1084,7 +1094,7 @@ rsset		: T_POP_RSSET uconst { sym_AddSet("_RS", $2); }
 rsreset		: T_POP_RSRESET { sym_AddSet("_RS", 0); }
 ;
 
-rs_uconst	: %empty {
+rs_uconst	: /* empty */ {
 			$$ = 1;
 		}
 		| uconst
@@ -1621,7 +1631,7 @@ strfmt_args	: string strfmt_va_args {
 		}
 ;
 
-strfmt_va_args	: %empty {
+strfmt_va_args	: /* empty */ {
 			initStrFmtArgList(&$$);
 		}
 		| strfmt_va_args T_COMMA const_no_str {
@@ -1645,7 +1655,7 @@ section		: T_POP_SECTION sectmod string T_COMMA sectiontype sectorg sectattrs {
 		}
 ;
 
-sectmod		: %empty { $$ = SECTION_NORMAL; }
+sectmod		: /* empty */ { $$ = SECTION_NORMAL; }
 		| T_POP_UNION { $$ = SECTION_UNION; }
 		| T_POP_FRAGMENT { $$ = SECTION_FRAGMENT; }
 ;
@@ -1660,7 +1670,7 @@ sectiontype	: T_SECT_WRAM0 { $$ = SECTTYPE_WRAM0; }
 		| T_SECT_OAM { $$ = SECTTYPE_OAM; }
 ;
 
-sectorg		: %empty { $$ = -1; }
+sectorg		: /* empty */ { $$ = -1; }
 		| T_LBRACK uconst T_RBRACK {
 			if ($2 < 0 || $2 >= 0x10000) {
 				error("Address $%x is not 16-bit\n", $2);
@@ -1671,7 +1681,7 @@ sectorg		: %empty { $$ = -1; }
 		}
 ;
 
-sectattrs	: %empty {
+sectattrs	: /* empty */ {
 			$$.alignment = 0;
 			$$.alignOfs = 0;
 			$$.bank = -1;
-- 
2.31.1

