--- tig-2.3.3.org/src/display.c	2018-03-10 03:45:51.796847979 +0000
+++ tig-2.3.3/src/display.c	2018-03-10 03:46:34.443314825 +0000
@@ -61,7 +61,7 @@
 }
 
 bool
-open_external_viewer(const char *argv[], const char *dir, bool silent, bool confirm, bool echo, bool refresh, const char *notice)
+open_external_viewer(const char *argv[], const char *dir, bool silent, bool confirm, bool echo, bool do_refresh, const char *notice)
 {
 	bool ok;
 
@@ -96,7 +96,7 @@
 		set_terminal_modes();
 	}
 
-	if (watch_update(WATCH_EVENT_AFTER_COMMAND) && refresh) {
+	if (watch_update(WATCH_EVENT_AFTER_COMMAND) && do_refresh) {
 		struct view *view;
 		int i;
 
