--- xinit-1.4.2/startx.cpp
+++ xinit-1.4.2.patched/startx.cpp
@@ -49,7 +49,7 @@
 
 userclientrc=$HOME/.xinitrc
 [ -f "${XINITRC}" ] && userclientrc="${XINITRC}"
-sysclientrc=XINITDIR/xinitrc
+sysclientrc=/etc/xinitrc
 
 userserverrc=$HOME/.xserverrc
 [ -f "${XSERVERRC}" ] && userclientrc="${XSERVERRC}"
@@ -253,7 +253,7 @@
     removelist=
 
     XCOMM set up default Xauth info for this machine
-    hostname=`uname -n`
+    hostname=localhost
 
     authdisplay=${display:-:0}
 #if defined(HAS_COOKIE_MAKER) && defined(MK_COOKIE)
